app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt

```diff
@@ -408,66 +447,37 @@ fun SearchBarListScreen(
-                  .testTag("SuggestionsMenuListScreen")) {
-            if (permissionHandler.getLocalisationPerm().collectAsState().value) {
-              DropdownMenuItem(
-                  text = {
-                    Text(stringResource(R.string.my_location), textAlign = TextAlign.Start)
-                  },
-                  modifier = Modifier.testTag("MyLocationSuggestionMenuItem"),
-                  contentPadding = PaddingValues(8.dp),
-                  onClick = {
-                    parkingViewModel.setCircleCenter(mapViewModel.userPosition.value)
-                    parkingViewModel.setMyLocation(true)
-                    showSuggestions.value = false
-                    textFieldValue.value = context.resources.getString(R.string.my_location)
-                    isTextFieldVisible.value = false
-                    focusManager.clearFocus()
-                  })
-            }
-
-            val seenNames = mutableSetOf<String>()
-            uniqueSuggestions.value =
-                listOfSuggestions.value.filter { suggestion ->
-                  val displayName =
-                      suggestion.suggestionFormatDisplayName(
-                          MAX_SUGGESTION_DISPLAY_NAME_LENGTH_LIST, Address.Mode.LIST)
-                  if (displayName in seenNames) {
-                    false
-                  } else {
-                    seenNames.add(displayName)
-                    true
-                  }
-                }
-
-            for (address in uniqueSuggestions.value.take(NUMBER_OF_SUGGESTIONS_FOR_MENU)) {
-              DropdownMenuItem(
-                  text = {
-                    Text(
-                        address.suggestionFormatDisplayName(
-                            MAX_SUGGESTION_DISPLAY_NAME_LENGTH_LIST, Address.Mode.LIST),
-                        textAlign = TextAlign.Start)
-                  },
-                  modifier = Modifier.testTag("SuggestionMenuItem${address.city}"),
-                  contentPadding = PaddingValues(8.dp),
-                  onClick = {
-                    parkingViewModel.setChosenLocation(address)
-                    showSuggestions.value = false
-                    parkingViewModel.setMyLocation(false)
-                    isTextFieldVisible.value = false
-                    textFieldValue.value =
-                        address.suggestionFormatDisplayName(
-                            MAX_SUGGESTION_DISPLAY_NAME_LENGTH_LIST, Address.Mode.LIST)
-                    focusManager.clearFocus()
-                  })
-            }
+    DropdownMenu(
+        expanded = showSuggestions.value,
+        onDismissRequest = { showSuggestions.value = false },
+        modifier =
+            Modifier.width(with(LocalDensity.current) { textFieldSize.value.width.toDp() })
+                .testTag("SuggestionsMenuListScreen")) {
+          // If there are no suggestions, display a message
```

Comment:
Thanks for inline comments like this that make the code clearer to follow!

Reply:
Thank you for noticing. I try to add as many comments to improve clarity